src/Operations.jl

```diff
@@ -643,25 +643,55 @@ function download_artifacts(ctx::Context, pkgs::Vector{PackageSpec}; platform::A
     return download_artifacts(ctx, pkg_roots; platform=platform, verbose=verbose)
 end
 
+"""
+    load_pkg_hook(hook_path::String)
+
+Given a path to a hook file (which should end in a function definition) include
+it into an anonymous module and return that function.
+"""
+function load_pkg_hook(hook_path::String)
+    m = Module(gensym())
+    f = Base.include(m, hook_path)::Function
+
+    # Because this function will most likely be used in an old world age,
+    # we will do our caller a solid and auto-invokelatest for it here:
+    return (args...) -> Base.invokelatest(f, args...)
+end
+
 function download_artifacts(ctx::Context, pkg_roots::Vector{String}; platform::AbstractPlatform=HostPlatform(),
                             verbose::Bool=false)
     # List of Artifacts.toml files that we're going to download from
-    artifacts_tomls = String[]
+    artifacts_tomls = Tuple{String,Function}[]
 
     for path in pkg_roots
         # Check to see if this package has an (Julia)Artifacts.toml
         for f in artifact_names
             artifacts_toml = joinpath(path, f)
             if isfile(artifacts_toml)
-                push!(artifacts_tomls, artifacts_toml)
+                hook_path = joinpath(path, ".pkg", "platform_augmentation_hook.jl")
+                if isfile(hook_path)
+                    try
```

Comment:
Consistently with the build script now throwing an exception if it fails, this should most likely also do so. If the script is so broken that it throws you don't want a tiny log message that is easy to miss when e.g. deploying it.

Personally, I think almost every try catch in the Artifact loading system should be removed since they make it easy for bugs to slip through, and then you run with a broken configuration. But I don't want to push that too hard.

Reply:
Sure, that's fair enough.